lib/puppet/provider/keycloak_client/kcadm.rb

```diff
@@ -221,6 +242,19 @@ def create
         raise Puppet::Error, "kcadm delete clients/#{resource[:id]}/optional-client-scopes/#{scope_id}: #{e.message}"
       end
     end
+    role = nil
+    if resource[:roles]
+      roles = get_client_roles(resource[:realm], resource[:id])
+      remove_roles = roles - resource[:roles]
```

Comment:
You shouldn't need to query roles again at this point, they will be stored in `@property_hash[:roles]` based on what was collected in `self.instances`.

Could make this:
```
remove_roles = @property_hash[:roles] - resource[:roles]
```